### PR TITLE
Use `package:extension_discovery` to detect DevTools extensions

### PR DESCRIPTION
**Pull request by kenzieschmoll**
Depends on https://github.com/dart-lang/tools/pull/129 landing and `package:extension_discovery` being published.

I have verified this package works as expected for the DevTools use case of detecting available extensions for a running application.
CC @jonasfj

### CODE REVIEW

File: packages/devtools_shared/lib/src/extensions/extension_manager.dart [hunk 1]
```diff
@@ -57,13 +58,16 @@ class ExtensionsManager {
...
+          path.join(
+            rootPath,
+            '.dart_tool',
+            'package_config.json',
+          ),
+        ),
+      );
       for (final extension in extensions) {
         final config = extension.config;
         if (config is! Map) {
```
Comment (jonasfj) [inline packages/devtools_shared/lib/src/extensions/extension_manager.dart:73]: nit: I'd suggest printing a warning to stderr, if for some reason the extension doesn't work.

Or at-least do it, if the extension is in the root package (`rootPath == extension.rootUri`. If I'm working on a package that contains an extension, I should probably be told if the extension doesn't work. Less important, if the extension is from a dependency.

Comment (kenzieschmoll) [reply packages/devtools_shared/lib/src/extensions/extension_manager.dart:73]: I can't think of a scenario where the root path would be equal to the `extension.rootUir`. For example, if I'm a package author for `package:foo`, my extension would live at `path/to/foo/extension/devtools`, but root uri would be for some app that uses foo (`path/to/some_app_that_depends_on_foo`), not `path/too/foo`

**Comment by kenzieschmoll** [comment]
Before landing this, we need to make sure that package:extension_discovery has been added as a DEP in the Dart SDK and that this package is rolled into google3.

File: packages/devtools_shared/lib/src/extensions/extension_manager.dart [hunk 2]
```diff
@@ -57,17 +58,31 @@ class ExtensionsManager {
...
+            path.join(
+              rootPath,
+              '.dart_tool',
+              'package_config.json',
+            ),
+          ),
+        );
+      } catch (e) {
+        print('[ERROR] `findExtensions` failed: $e');
+        extensions = <Extension>[];
```
Comment (polina-c) [inline packages/devtools_shared/lib/src/extensions/extension_manager.dart:75]: Do we want to report thid to analytics?

File: packages/devtools_shared/lib/src/extensions/extension_manager.dart [hunk 3]
```diff
@@ -57,17 +58,31 @@ class ExtensionsManager {
...
+      }
+
       for (final extension in extensions) {
         final config = extension.config;
         if (config is! Map) {
           // Fail gracefully. Invalid content in the extension's config.json.
+          print(
+            '[WARNING] invalid config.json content for ${extension.package}:\n'
+            '$config',
```
Comment (polina-c) [inline packages/devtools_shared/lib/src/extensions/extension_manager.dart:84]: And this?

**Review by polina-c — APPROVED** [review]
Couple comments and LGTM.

**Pull request merged**